internal/metrics/http_middleware.go

```diff
@@ -97,15 +97,7 @@ func HTTPMiddleware(config HTTPMiddlewareConfig) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
-
-		switch {
-		case isSpdyRequest(r):
-			ctx = context.WithValue(ctx, contextKey{}, requestTypeSPDY)
-		case wsstream.IsWebSocketRequest(r):
-			ctx = context.WithValue(ctx, contextKey{}, requestTypeWebsocket)
-		default:
-			ctx = context.WithValue(ctx, contextKey{}, requestTypeHTTP)
-		}
+		ctx = withRequestType(ctx, r)
 
 		base.ServeHTTP(w, r.WithContext(ctx))
```

Comment:
Would it be simpler to get a `http.Request` and return a new request with the enriched context?

Reply:
[`contextcheck`](https://github.com/kkHAIKE/contextcheck) lint won't allow it. I am getting
```
 Non-inherited new context, use function like `context.WithXXX` or `r.Context` instead (contextcheck)
		base.ServeHTTP(w, r.WithContext(ctx))
		                               ^
```

Reply:
Yes! thanks for the suggestions!